plugins/de.cau.cs.kieler.klighd.piccolo/src/de/cau/cs/kieler/klighd/piccolo/internal/KlighdSWTGraphicsImpl.java

```diff
@@ -437,7 +437,7 @@ public FontData getFontData() {
     protected static final HashMap<FontData, Font> FONT_CACHE = new HashMap<FontData, Font>();
 
     private boolean cacheFonts = true;
-    private List<org.eclipse.swt.graphics.Font> temporaryFonts = null;
+    private final List<Font> temporaryFonts = new ArrayList<>();
```

Comment:
If this field may be null all accesses have to be guarded with a null check, but not all were guarded.
Instead of adding more guards it's simpler to just initialize the field immediately. And creating an `ArrayList` without arguments is cheap since the backing array is only created on the first addition of an element. If nothing is added, it's just the 'empty' object.